gabbi/case.py

```diff
@@ -437,6 +412,31 @@ def _test_status(self, expected_status, observed_status):
 
         self.assert_in_or_print_output(observed_status, statii)
 
+    def _update_query_params(self, original_query_string, query_params):
+        """Update a query string from query_params dict.
+
+        An OrderedDict is used to allow easier testing and greater
```

Comment:
I always feel bad when adjusting runtime code merely for the sake of testing. I suppose this is necessary because you (quite reasonably) want to use string comparisons in the tests below, rather than parsing the query string and verifying the individual parts thereof?